reconcile/utils/terrascript_client.py

```diff
@@ -4073,7 +4093,63 @@ def populate_tf_resource_secrets_manager(self, resource, namespace_info):
 
         self.add_resources(account, tf_resources)
 
-    def populate_tf_resource_asg(self, resource, namespace_info):
+    def get_asg_image_id(self, image, existing_secrets, output_prefix,
+                         account, region):
+        image_id = image.get('id')
+        if image_id:
+            return image_id, ''
+
+        image_repo = image.get('repo')
+        tag_name = image_repo.get('tag_name')
+        url = image_repo.get('url')
+        ref = image_repo.get('ref')
```

Comment:
As you mentioned that these fields are mandatory in the schema, remove `get` here:
```suggestion
        image_repo = image['repo']
        tag_name = image_repo['tag_name']
        url = image_repo['url']
        ref = image_repo['ref']
```